File: src/main/scala/click/seichi/eategg/config/EnabledWorlds.scala
package click.seichi.eategg.config

import org.bukkit.plugin.java.JavaPlugin

import java.util.Collections
import scala.collection.mutable
import scala.jdk.CollectionConverters._

object EnabledWorlds {
  private final val ConfigPath = "enabled-worlds"
  private val worlds: mutable.Set[String] = mutable.Set.empty

  def load()(implicit instance: JavaPlugin): Unit = {
    worlds.clear()

    instance.getConfig.getStringList(ConfigPath).asScala.map(_.toLowerCase).foreach(worlds.add)
  }

  def get(): Set[String] = worlds.toSet

  def contains(worldName: String): Boolean = get().contains(worldName.toLowerCase)
}

File: src/main/java/click/seichi/eategg/config/EnabledWorlds.java
package click.seichi.eategg.config;

import java.util.HashSet;
import java.util.Locale;
import java.util.Set;
import org.bukkit.plugin.java.JavaPlugin;

public final class EnabledWorlds {
  private static final String CONFIG_PATH = "enabled-worlds";
  private static final Set<String> WORLDS = new HashSet<>();

  private EnabledWorlds() {}

  public static void load(JavaPlugin instance) {
    WORLDS.clear();
    for (String world : instance.getConfig().getStringList(CONFIG_PATH)) {
      WORLDS.add(world.toLowerCase(Locale.ROOT));
    }
  }

  public static Set<String> get() {
    return Set.copyOf(WORLDS);
  }

  public static boolean contains(String worldName) {
    return WORLDS.contains(worldName.toLowerCase(Locale.ROOT));
  }
}
